File: src/bj11047/Main.java
package bj11047;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input[] = br.readLine().split(" ");
        int[] coin=new int[Integer.parseInt(input[0])];
        int result=0;
        int wallet=Integer.parseInt(input[1]);
        for(int i=0; i<Integer.parseInt(input[0]); i++){
            coin[i]=Integer.parseInt(br.readLine());
        }
        while(wallet>0) {
            int index=0;
            for (int i = 0; i < Integer.parseInt(input[0]); i++) {
                if (wallet < coin[i]) {
                    index=i-1;
                    break;
                } else if(i == Integer.parseInt(input[0])-1&&wallet>coin[i]){
                    index=i;
                    break;
                }
            }
            int num=wallet/coin[index];
            wallet-=coin[index]*num;
            result+=num;
        }
        System.out.println(result);
    }
}

File: kotlin/src/main/kotlin/bj11047/Main.kt
package bj11047

import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val nk = br.readLine().split(" ").map { it.toInt() }
    val n = nk[0]
    var k = nk[1]
    val coins = IntArray(n)
    for (i in 0 until n) {
        coins[i] = br.readLine().toInt()
    }
    var result = 0
    for (i in n - 1 downTo 0) {
        result += k / coins[i]
        k -= k / coins[i] * coins[i]
    }
    bw.write("$result")
    br.close()
    bw.close()
}
